components/raftstore/src/store/worker/pd.rs

```diff
@@ -728,78 +715,72 @@ where
             .set(available as i64);
 
         let router = self.router.clone();
-        let f = Compat::new(self.pd_client.store_heartbeat(stats))
-            .map_err(|e| {
-                error!("store heartbeat failed"; "err" => ?e);
-            })
-            .map(move |mut resp| {
+        let resp = self.pd_client.store_heartbeat(stats).map_err(|e| {
+            error!("store heartbeat failed"; "err" => ?e);
```

Comment:
Better to move it to async block. It's strange to handle the result in different places.